### PR TITLE
Add layer for Semantic Web formats

### PR DESCRIPTION
**Pull request by atextor**
- Syntax highlighting for N3, Turtle and SPARQL formats
- Company mode for SPARQL
- Execution of SPARQL queries against SPARQL HTTP endpoint

### CODE REVIEW

File: layers/+lang/semweb/packages.el [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    :defer t))
+
+(defun semweb/init-sparql-mode ()
+  (with-eval-after-load 'company
+    (use-package sparql-mode
+      :mode ("\\.\\(sparql\\|rq\\)\\'" . sparql-mode)
+      :defer t
+      :init
+      (progn
+        (company-mode-on)
```
Comment (TheBB) [inline layers/+lang/semweb/packages.el]: Please configure company like it should be done (see other layers, for example python).

File: layers/+lang/semweb/packages.el [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        (add-hook 'sparql-mode-hook 'company-mode)
+        (with-eval-after-load 'org
+          (org-babel-do-load-languages 'org-babel-do-load-languages '((sparql . t))))
```
Comment (TheBB) [inline layers/+lang/semweb/packages.el]: Should be in an org package config, see e.g. https://github.com/syl20bnr/spacemacs/blob/develop/layers/+lang/clojure/packages.el#L295

File: layers/+lang/semweb/packages.el [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+;; Author: Andreas Textor <mail@atextor.de>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+(setq semweb-packages
+  '(
+    company
```
Comment (TheBB) [inline layers/+lang/semweb/packages.el]: Don't add this package unless you have a package init function for it.

File: layers/+lang/semweb/packages.el [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+(defun semweb/init-ttl-mode ()
+  (use-package ttl-mode
+    :mode ("\\.\\(ttl\\|n3\\)\\'" . ttl-mode)
+    :defer t))
+
+(defun semweb/init-sparql-mode ()
+  (with-eval-after-load 'company
+    (use-package sparql-mode
+      :mode ("\\.\\(sparql\\|rq\\)\\'" . sparql-mode)
+      :defer t
```
Comment (TheBB) [inline layers/+lang/semweb/packages.el]: `:mode` implies `:defer t`

File: layers/+lang/semweb/packages.el [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  '(
+    company
+    ttl-mode
+    sparql-mode
+    ))
+
+(defun semweb/init-ttl-mode ()
+  (use-package ttl-mode
+    :mode ("\\.\\(ttl\\|n3\\)\\'" . ttl-mode)
+    :defer t))
```
Comment (TheBB) [inline layers/+lang/semweb/packages.el]: `:mode` implies `:defer t`

**Comment by atextor** [comment]
Thank you for the quick reply! I applied the requested changes except for the `company-mode-on`: I found that `sparql-mode`, unlike other modes, writes to a variable during initialisation that is not defined unless company-mode is enabled (`company-keywords-alist`). This is the only way I could make it work.

**Comment by TheBB** [comment]
We should get sparql-mode to fix their shit then.

**Comment by TheBB** [comment]
https://github.com/ljos/sparql-mode/issues/52

**Comment by TheBB** [comment]
My PR was merged, so this should be doable on our side now.

**Comment by atextor** [comment]
With your fix in place, I could remove the problematic part. Works like a charm.

**Comment by atextor** [comment]
Are there any more changes necessary?

**Comment by Stebalien** [comment]
Personally, I'd call this layer `semantic-web`. It's not *that* hard to type :wink:.

**Comment by TheBB** [comment]
Thanks! Cherry-picked in develop. You can safely delete your branch.

I took @Stebalien's advice and renamed it to semantic-web.

**Pull request closed without merging**